### PR TITLE
Add --reload-browser flag to conditionally run browser-sync

### LINKED ISSUE #408: Add `lucky watch --reload-browser`
As discussed in [this pull request lucky-cli](https://github.com/luckyframework/lucky_cli/pull/175#issuecomment-378288754). The `--reload-browser` flag will add browser-sync to the `lucky watch` task.

### PR DESCRIPTION
**Pull request by mikeeus**
Closes #408.

I used OptionParser to set a `@reload_browser` property and use that to conditionally start/reload browser-sync in the `create_app_processes` method.

### CODE REVIEW

File: tasks/watch.cr [hunk 1]
```diff
@@ -38,10 +41,12 @@ module Sentry
...
-      if successful_compilations == 1
-        start_browsersync
-      else
-        reload_browsersync
+      if @reload_browser
+        if successful_compilations == 1
+          start_browsersync
+        else
+          reload_browsersync
+        end
```
Comment (paulcsmith) [inline tasks/watch.cr]: I think extracting a `reload_or_start_browser_sync` method would be nice so this is cleaned up a bit

File: tasks/watch.cr [hunk 2]
```diff
@@ -15,11 +16,13 @@ module Sentry
     getter app_processes = [] of Process
     property successful_compilations
     property app_built
+    property reload_browser
```
Comment (paulcsmith) [inline tasks/watch.cr]: If you do `property?` you can then call `reload_browser?` which reads a bit better IMO

File: tasks/watch.cr [hunk 3]
```diff
@@ -38,10 +41,12 @@ module Sentry
       end
 
       self.successful_compilations += 1
-      if successful_compilations == 1
-        start_browsersync
-      else
-        reload_browsersync
+      if @reload_browser
```
Comment (paulcsmith) [inline tasks/watch.cr]: Then you can use `if reload_browser?`

File: tasks/watch.cr (hunk 2 above)
Comment (mikeeus) [reply tasks/watch.cr]: That's neat, it definitely reads better

**Pull request merged**